src/runtime/pkg/katautils/config.go

```diff
@@ -607,6 +607,10 @@ func (h hypervisor) extraMonitorSocket() (govmmQemu.MonitorProtocol, error) {
 func (h hypervisor) sharedFS() (string, error) {
 	supportedSharedFS := []string{config.Virtio9P, config.VirtioFS, config.VirtioFSNydus, config.NoSharedFS}
 
+	if h.ConfidentialGuest && h.SharedFS != config.NoSharedFS {
```

Comment:
@fidencio , I would prefer enforcing required values in a single place - in the Kata Agent, because it's part of the CoCo TCB. That way we'd have to relax the "policy" in a single place for potential debug experiments - instead of having to relax both the Shim and the Agent sides.

Reply:
> @fidencio , I would prefer enforcing required values in a single place - in the Kata Agent

I kinda agree, even if we could fail earlier with the patch on the runtime side.

Keep in mind that currently there's no way for the agent to understand whether it's running inside a TEE or not (or, IOW, no explicit way to define when the "policy" should be enforced).  There are several implicit ways to do that, though.

Reply:
Should we add detection capability for the agent to know if CC or not? Like checking for /dev files? AMD has sev guest what about TDX or CCA?

Reply:
> Should we add detection capability for the agent to know if CC or not? Like checking for /dev files? AMD has sev guest what about TDX or CCA?

If we take this approach, I'd be in favour of using what we have on `cpuid` (see: https://github.com/klauspost/cpuid).  There we already have ways to detect TDX guest virtualised on Linux and Hyper-V.  Expanding that to support SEV / SNP / SE / CCA would be a good thing.  Mind that "expanding that", in our case, would be writting a similar and very specific crate in rust, which is easily doable if we decide to take this path.

Reply:
Agent Policy and/or Init Data are expected to provide integrity-protected configuration similar to this one.

So, unless we have use cases that don't use either Agent Policy or Init Data, it would be a shame to add a third Policy component for this.

However, if you would like to hard-code those restrictions instead of allowing run time configuration, I guess we could compile out the relevant Agent code, similar to what @fitzthum said.

Reply:
> Agent Policy and/or Init Data are expected to provide integrity-protected configuration similar to this one.

I'd rather use Agent Policy for this, to be honest.

I don't entirely dislike the idea of compiling out the relevant code, but then hey, yet another agent version to do CI on, and I'd rather not have to deal with this.

Reply:
So, action item for me for the coming days, explore the current policies and check whether I can use 9p / virtiofs mounts with with the current policies, and then I will report it back here in the thread.

Reply:
@danmihai1, JFYI, policy, as it is Today, prevents the following issue:
* User deploys nydus-snapshotter
* User configures containerd to use nydus-snapshotter
* User deploys Kata Containers and sets its configuration with `shared_fs = none`
* The infrastructure owner changes the Kata Containers configuration to use `shared_fs = 9p`
* The infrastructure owner sets containerd to not use nydus as the Kata Containers snapshotter

With policy generated the pod will work with `shared_fs = none` but the container will not be created with `shared_fs = 9p`.

I'm closing this one, and I will open a new issue so we can discuss what's the best approach to, maybe, have this specific check as part of the default policy set in the rootfs.

Reply:
Sounds good, thanks @fidencio !